Support missing message fields

Doc94 opened the pull request:
**Description:** Its a WIP about little changes in discord messages (WIP Because its working but not document) and anothe changes for discord features.

Reference: Tests in servers and https://github.com/discordapp/discord-api-docs/pull/1051

Doc94 commented:
I have a little question about add support to "message_reference" in the new type of message (follow channel) this param show the origin of follow.
Example using Discord API.
`[
    {
        "attachments": [],
        "tts": false,
        "embeds": [],
        "timestamp": "2019-08-09T15:45:32.088000+00:00",
        "mention_everyone": false,
        "id": "609411953697357825",
        "pinned": false,
        "edited_timestamp": null,
        "message_reference": {
            "channel_id": "326437277217849345",
            "guild_id": "322850917248663552"
        },
        "author": {
            "username": "Doc",
            "discriminator": "4444",
            "id": "216316781483130880",
            "avatar": "9e56b7acf4f1dafa7b75d5aec4b2deef"
        },
        "mention_roles": [],
        "content": "Official Fortnite #twitter-feed",
        "channel_id": "609411844272029716",
        "mentions": [],
        "type": 12
    }
]`

Shadorc commented on core/src/main/java/discord4j/core/object/entity/Message.java:
```diff
@@ -477,7 +477,10 @@ public int hashCode() {
+        USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_3(11),
+
+        /** A message created when an user follow a channel from another guild into specific channel (<a href="https://support.discordapp.com/hc/en-us/articles/360028384531-Server-Following-FAQ">Server Following</a>). */
+        CHANNEL_FOLLOW_ADD(11);
```
I think it should be 12

Doc94 replied:
oh.....
i know i forget something.

Doc94 commented:
Of commit bf1e0d7 im not secure about the flags, with integer optionals.
In case of messagereference im create this out of Message (dont know if this fine).
(Im try to create a class like Type for the 3 flags but dont know how to work the "<<")

Doc94 commented:
A example of Message::toString with flags https://hasteb.in/yavafoxe.kotlin

Shadorc commented on core/src/main/java/discord4j/core/object/entity/Guild.java:
```diff
@@ -398,6 +398,8 @@ public ContentFilterLevel getContentFilterLevel() {
     /**
      * Gets the enabled guild features.
+     * <br>
+     * You can see the avaible <a href="https://discordapp.com/developers/docs/resources/guild#guild-object-guild-features">guild features</a>
```
the avaible

Doc94 commented:
Fixed supress method, added a "deprecated" for future changes by discord (but dont think change this... maybe response with a message edited and not empty).

quanticc commented:
I merged this while setting some API as Experimental.

Since this PR was a bit stale, @Doc94 Can you give us some feedback if you notice any issues? Thanks in advance.
